drivers/watchdog/wdt_nrfx.c

```diff
@@ -21,41 +21,40 @@ struct wdt_nrfx_data {
 
 struct wdt_nrfx_config {
 	nrfx_wdt_t	  wdt;
-	nrfx_wdt_config_t config;
 };
 
 static int wdt_nrf_setup(const struct device *dev, uint8_t options)
 {
-	const struct wdt_nrfx_config *config = dev->config;
-	struct wdt_nrfx_data *data = dev->data;
 	nrfx_err_t err_code;
+	nrfx_wdt_config_t config;
+	const nrfx_wdt_t *wdt = &dev->config.wdt;
 
 	/* Activate all available options. Run in all cases. */
-	config->config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK |
+	config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK |
 #if NRF_WDT_HAS_STOP
-				   NRF_WDT_BEHAVIOUR_STOP_ENABLE_MASK |
+			   NRF_WDT_BEHAVIOUR_STOP_ENABLE_MASK |
 #endif
-				   NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
+			   NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
 
 	/* Deactivate running in sleep mode. */
 	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
-		config->config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK;
+		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK;
 	}
 
 	/* Deactivate running when debugger is attached. */
 	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
-		config->config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
+		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
 	}
```

Comment:
I'd propose to refactor this to:
```
	if (NRF_WDT_HAS_STOP) {
		config.behaviour |= NRF_WDT_BEHAVIOUR_STOP_ENABLE_MASK;
	}
	if (!(options & WDT_OPT_PAUSE_IN_SLEEP)) {
		config.behaviour |= NRF_WDT_BEHAVIOUR_RUN_SLEEP_MASK;
	}
	if (!(options & WDT_OPT_PAUSE_HALTED_BY_DBG)) {
		config.behaviour |= NRF_WDT_BEHAVIOUR_RUN_HALT_MASK;
	}
```

Reply:
Unfortunately, the first statement can't be made this way.  The `NRF_WDT_BEHAVIOUR_STOP_ENABLE_MASK` symbol is defined in nrfx under preprocessor guards which causes compilation to fail if your idea is used.